### DISCUSSION TITLE
longhorn-csi-plugin + engine-image-ei  + csi from helm chart appears to be running on all nodes

### DISCUSSION THREAD

**Discussion by ehudyonasi** [longhorn/longhorn#4740] [Q&A] [answered]
Hello,
I've tried using the helm chart to install longhorn on specific labels using flux:
  values:
      longhornManager:
        nodeSelector:
          "longhorn": "true"
      longhornDriver:
        nodeSelector:
          "longhorn": "true"
      longhornUI:
        nodeSelector:
          "longhorn": "true"

I see that the longhorn manager, ui and driver are running correctly on those nodes, but the rest are just occupying my worker nodes.. 
Am I missing anything?

**Comment by ehudyonasi** [comment]
okay I can see that there was an issue regarding it: [https://github.com/longhorn/longhorn/issues/808](url)
and it is done by longhorn code...

**Comment by weizhe0422** [comment]
Would you mind to provide more information about the scenario?

IIUC, the POD will be launched on the node with the highest score according to the kube-scheduler.

**Reply by ehudyonasi** [reply]
Sure,
I would like not to spawn the workloads on my worker nodes to save the cpu / memory on them as they will not be used for that purpose.
But I can see that all the rest of the components are scheduled without the NodeSelector. such components are:
engine-image-ei-*
longhorn-csi-plugin
all csi-* (attacher, provisioner, resizer, snapshooter)

**Comment by PhanLe1010** [comment, marked as answer]
You are missing one setting, `Set node selector for system managed components`, please see this doc for full details https://longhorn.io/docs/1.3.2/advanced-resources/deploy/node-selector/

**Reply by ehudyonasi** [reply]
This is what I did :)
as you can read my first comment with the nodeSelectors, but sadly inside longhorn code they deploy the csi plugin and engine **without node selection** and it does not work if I edit them manually, the code will just return them back to previous state.
By the issue I mentioned I can see that the main reason is because the longhorn developers would like to keep that in control instead of giving the choice to the users because of upgrade issues later on..

**Reply by PhanLe1010** [reply]
Sorry, I got confused. So was the issue gone after setting `Set node selector for system managed components`?

**Reply by ehudyonasi** [reply]
No, it is the same.

**Reply by PhanLe1010** [reply]
It is weird. Can you upload a support bundle?

**Reply by ehudyonasi** [reply]
Well after looking at it in the last couple of days, I can actually see it is selecting my desired nodes, thanks for your help!
